File: Dev/CS/Mascaret/Mascaret/BEHAVE/SimpleBehaviorExecution.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public class SimpleBehaviorExecution : AgentBehaviorExecution
    {

        private double interval;
        public double Interval
        {
            get { return interval; }
            set { interval = value; }
        }

        public SimpleBehaviorExecution(Behavior specif, InstanceSpecification host, Dictionary<String, ValueSpecification> p)
            : base(specif, host, p)
        {

        }

        public SimpleBehaviorExecution()
        {
        }

        public override double execute(double dt)
        {
            action();
            if (!done())
                return interval;
            else return 0;
        }

        public override void action()
        {
            throw new NotImplementedException();
        }

        public override bool done()
        {
            throw new NotImplementedException();
        }
    }
}

File: Dev/C++/Mascaret/src/BEHAVE/SimpleBehaviorExecution.cpp
#include "BEHAVE/SimpleBehaviorExecution.h"

namespace BEHAVE
{

SimpleBehaviorExecution::SimpleBehaviorExecution(shared_ptr<Behavior> specif,shared_ptr<InstanceSpecification> host, const Parameters& p) : AgentBehaviorExecution(specif,host,p), _interval(0.2)
{}

SimpleBehaviorExecution::~SimpleBehaviorExecution()
{}

double SimpleBehaviorExecution::execute(double dt)
{
	action();
	if (!done()) return _interval;
	else return 0;
}

}
